ideascube/configuration/management/commands/config.py

```diff
@@ -0,0 +1,153 @@
+
+        list = subs.add_parser(
+            'list', help='List configuration namespaces and keys')
+        list.add_argument(
+            'namespace', nargs='?',
+            help='Only list configuration keys for this namespace')
+        list.set_defaults(func=self.list_configs)
+
+        reset = subs.add_parser(
+            'reset', help='Reset a configuration option to its default value')
+        reset.add_argument('namespace', help='The configuration namespace')
+        reset.add_argument('key', help='The configuration key')
+        reset.set_defaults(func=self.reset_config)
+
+        set = subs.add_parser(
+            'set', help='Set the value of a configuration option')
+        set.add_argument('namespace', help='The configuration namespace')
+        set.add_argument('key', help='The configuration key')
+        set.add_argument('value', help='The new value')
+        set.set_defaults(func=self.set_config)
+
+        self.parser = parser
+
+    def handle(self, *_, **options):
+        if 'func' not in options:
+            self.parser.print_help()
+            self.parser.exit(1)
+
+        options['func'](options)
+
+    def _evaluate_value(self, value):
+        if value in ('true', 'false'):
+            # Special case booleans for a better UX
+            value = value.capitalize()
+
+        try:
+            return ast.literal_eval(value)
+
+        except (SyntaxError, ValueError):
+            # Admin probably wanted a string and didn't use enough quotes
+            # examples:
+            #     $ ideascube config set server site-name thename
+            #     $ ideascube config set server site-name 'the name'
+            return str(value)
+
+    def get_config(self, options):
+        namespace = options['namespace']
+        key = options['key']
+
+        try:
+            value = get_config(namespace, key)
+
+        except (NoSuchConfigurationNamespaceError,
+                NoSuchConfigurationKeyError) as e:
+            raise CommandError(e)
+
+        print('%r' % value)
+
+    def help_for_config(self, options):
+        data = {
+            'namespace': options['namespace'],
+            'key': options['key'],
+        }
+
+        try:
+            data.update(get_config_data(data['namespace'], data['key']))
+
+        except (NoSuchConfigurationNamespaceError,
+                NoSuchConfigurationKeyError) as e:
+            raise CommandError(e)
+
+        print(
+            '# {namespace} {key}\n\n'
+            '{summary}\n\n'
+            '* type: {pretty_type}\n'
+            '* default value: {default!r}'.format(**data))
+
+    def list_configs(self, options):
+        namespace = options['namespace']
+
+        if namespace is None:
+            for namespace in get_all_namespaces():
+                print(namespace)
+
+            return
+
+        try:
+            for key in get_namespaced_configs(namespace):
+                print("%s %s" % (namespace, key))
```

Comment:
> the first thing I've wanted to see is "how is the config here overall"

Now this is a very interesting one, and I hadn't thought about it. Some kind of full report of the whole configuration of the box.

This is why I've been pushing back by the way: to get an actual use-case rather than merely a UI suggestion. :wink: 

I'm totally going to implement this today, because it's such a useful feature and I'm kind of sad I didn't think about it. Plus, it could probably be added to the CLI without interfering with any of the rest.

Thanks! :smiley: